kinematics.go

```diff
@@ -128,12 +113,11 @@ var MaxInverseKinematicIteration int = 50
 // InverseKinematics calculates joint angles to achieve an XyzWxyz end effector
 // position given the desired XyzWxyz coordinates and the robotic arm
 // parameters.
-func InverseKinematics(desiredEndEffector XyzWxyz, dhParameters DhParameters) (StepperTheta, error) {
-	thetasInit := StepperTheta{0, 0, 0, 0, 0, 0}
+func InverseKinematics(desiredEndEffector XyzWxyz, dhParameters DhParameters) ([]float64, error) {
+	thetasInit := []float64{0, 0, 0, 0, 0, 0}
 	// Initialize an objective function for the optimization problem
 	objectiveFunction := func(s []float64) float64 {
-		stepperThetaTest := StepperTheta{s[0], s[1], s[2], s[3], s[4], s[5]}
-		currentEndEffector := ForwardKinematics(stepperThetaTest, dhParameters)
+		currentEndEffector := ForwardKinematics(s, dhParameters)
 
 		// Get XYZ offsets
 		xOffset := desiredEndEffector.X - currentEndEffector.X
```

Comment:
Perhaps the `XyzWxyz` could support a difference operator for position?

Reply:
We could also maybe support the dot product for quaternions? We could use a library for the Quaternion, for example:
https://github.com/westphae/quaternion
We could also maybe use Gonum to back the quaternion and XYZ so we could get dot product built in.

Reply:
> We could also maybe support the dot product for quaternions?

Why? Not trying to be pedantic, but I have very little idea what the hell quaternions are really used for, so I'm not sure what the dot product does for you.

Reply:
So Quaternions are 4 dimensional vectors, that means that the dot product can be used to calculate the "angle" between the two quaternions. That's pretty useful to know how far quaternions are away from eachother.